### ISSUE TITLE
Feature request: Additional services, especially lastfm

### ISSUE THREAD

**Issue by inktrap** [hmhrex/Grav-MusicCard#4]
How hard would it be to add additional services? I am thinking of last.fm here. More sites [are listed here, for example](https://alternativeto.net/software/spotify/).

A lot of music is also on e.g. youtube.  However, I guess you want to keep musiccard strictly limited to music. The outdated [MediaEmbed](https://github.com/sommerregen/grav-plugin-mediaembed)-Plugin supported youtube and vimeo. However, given the fact that mediaembed is the successor of embed and videoembed, it seems complicated/a lot of work to support multiple video hosting sites. After searching around a little bit I found the  [vimeo plugin](https://github.com/cworreschk/grav-plugin-vimeo) and the [youtube plugin](https://github.com/getgrav/grav-plugin-youtube), so musiccard, vimeo and youtube could work well with each other.

**Comment by hmhrex** [comment]
MusicCard was originally written to use in a blog setting where you could quickly see metadata and play the song or album (from multiple sources) when it was referenced.

There were a few reasons I narrowed down the choices to Bandcamp, Spotify and SoundCloud:
* Last.fm does have metadata (including artwork), but has no audio source.
* YouTube doesn't have consistent metadata, and I believe separating audio from video (not displaying the video on the card) is against their ToS.
* Vimeo is in the same boat as YouTube.

Here are some that I wouldn't mind supporting in the future if there was interest:
* Apple Music
* Deezer
* Tidal

However, this project isn't really active as there hasn't been a great amount of interest. If you want to fork it, or create pull requests on it, be my guest!

**Comment by inktrap** [comment]
I'll keep that in mind and will check out deezer and tidal first. :) Mmh, last.fm indeed only includes the info :(

**Comment by hmhrex** [comment]
@inktrap That being said, if you want to fork this project and use it as a base to make something more like MediaEmbed, you are more than welcome 😊

**Comment by inktrap** [comment]
Yes, i read that between the lines ;) however I want to finish other projects first and I'll see how far I get with the plugins mentioned in the first comment. ;) Thanks though!
